pkg/services/provisioning/datasources/datasources.go

```diff
@@ -40,6 +40,13 @@ func Provision(ctx context.Context, configDirectory string, dsService BaseDataSo
 	return dc.applyChanges(ctx, configDirectory)
 }
 
+// GetCacheConfigs scans a directory for provisioning config files
+// and returns cache configs of the the datasources in those files.
+func GetCacheConfigs(ctx context.Context, configDirectory string, dsService BaseDataSourceService, correlationsStore CorrelationsStore, orgService org.Service) (*DatasourceCachingInfo, error) {
+	dc := newDatasourceProvisioner(log.New("provisioning.datasources"), dsService, correlationsStore, orgService)
```

Comment:
nit: you didn't introduce passing the logger into this function, but now that we are calling it in two places I think we should refactor it. the `newDatasourceProvisioner` method should create the logger instead of the caller having to.